src/compiler/commandLineParser.ts

```diff
@@ -1790,7 +1790,26 @@ namespace ts {
             return returnValue ? {} : undefined;
         }
 
-        return convertPropertyValueToJson(sourceFile.statements[0].expression, knownRootOptions);
+        const rootExpression: Expression = sourceFile.statements[0].expression;
+        if (rootExpression.kind !== SyntaxKind.ObjectLiteralExpression) {

```

Comment:
This is common functionality for json modules as well as tsconfig. 
The way tsconfig ensures its object is by passing knownRootOptions? This should not break module which has array.

Reply:
It does break, because there’s no name for the root of the object itself. We try to create an error message that says `Compiler option '{0}' requires a value of type 'object'` but there’s nothing to substitute for `{0}`, and the error doesn’t make much sense calling the root object itself a “compiler option.” So I think there should be dedicated logic for the root somewhere, because there should be a dedicated error message.

> This is common functionality for json modules as well as tsconfig.

Hah, ok, the failing test shows this. Very surprising since this is in `commandLineParser.ts`, but I’ll find another place for it.

Reply:
> Hah, ok, the failing test shows this. Very surprising since this is in commandLineParser.ts

that’s just from fact that we wrote this for tsconfig first and then added support for json modules